docs/library-distribution.md

```diff
@@ -0,0 +1,99 @@
+# Library Distribution
+
+This document describes the considerations and recommendations for using the
+OpenTelemetry-cpp library.
+
+## Definitions
+
+### Binary vs Source Distribution
+
+The OpenTelemetry-cpp libraries (API and SDK) and any library that depends on
+them are expected to be distributed as either "source" or "binary" artefacts.
+This document defines these distribution models as:
+
+- **Source** - The source code is shipped as-is and can be built as desired
+- **Binary** - The library and/or its dependants may be distributed as pre-built
+binaries which can be built using different toolchains or compilers.
+
+This document will focus on the specific challenges of the "binary" distribution
+model.
+
+### Library Binding Model
+
+The library and its dependants can be distributed and bound (loaded) in multiple
+ways which are defined as follows:
+
+- **Static Linkage** - The library is distributed as a compiled artefact (`.a`
+extension on Unix-like systems) which is linked and bound at compile-time.
+- **Dynamic Linkage with Early Binding** - The library is distributed as a
+compiled artefact (`.so` on Unix-like systems), and its binding is defined at
+compile-time. It is loaded by the dynamic linker when the application/library
+depending on it is started/loaded.
+- **Dynamic Linkage with Late Binding** - The library is distributed as a 
+compiled artefact (`.so` on Unix-like systems), only its interface is known at
+compile-time. It is loaded dynamically (using `dlopen` on POSIX).
+
+*NOTE: A library may itself link in other libraries based on any of the above.*
+
+### Components
+
+The following components will be considered for analysing the impact of
+different binding models and the recommendations.
+
+- **OpenTelemetry Libraries** - The libraries produced from this project,
+the API library will contain some globals/singletons such as `TracerProvider`.
```

Comment:
The API library will also contain implementations of distributed tracing and basic in-proc context propagation.

Reply:
I've reworded this to include "minimal implementation" as is the terminology used in the reference documentation in the specification. I wanted to focus on the characteristics relevant to the context (i.e. compiled binary artifacts) rather than thoroughly describe the content, but happy to change